### PR TITLE
Annotation work /File Cleaning part 2

### PR DESCRIPTION
**Pull request by MrRowey**
This Tackles the following files
- EngineerManager.lua
- Entity.lua
- FactioryBuilderManager.lua
- MapUtilities.lua
- MarkerUtilities.lua
- MatchState.lua
- NukeDamage.lua
- PlatoonFormManager.lua
- defaultweapons.lua

### CODE REVIEW

File: lua/sim/defaultweapons.lua [hunk 1]
```diff
@@ -1122,11 +1164,15 @@ OverchargeWeapon = Class(DefaultProjectileWeapon) {
...
 
+    ---@param self OverchargeWeapon
+    ---@return boolean
     HasEnergy = function(self)
         return self.Brain:GetEconomyStored('ENERGY') >= self.EnergyRequired
     end,
 
     -- Can we use the OC weapon?
+    ---@param self OverchargeWeapon
+    ---@return false
```
Comment (Hdt80bro) [inline lua/sim/defaultweapons.lua:1175]: I think you meant `@return boolean`

File: lua/sim/defaultweapons.lua [hunk 2]
```diff
@@ -1135,10 +1181,13 @@ OverchargeWeapon = Class(DefaultProjectileWeapon) {
             unit:IsUnitState('Upgrading')
     end,
 
+    ---@param self OverchargeWeapon
     StartEconomyDrain = function(self) -- OverchargeWeapon drains energy on impact
     end,
 
     -- Returns true if the unit is doing something that shouldn't allow any weapon fire
+    ---@param self OverchargeWeapon
+    ---@return true
```
Comment (Hdt80bro) [inline lua/sim/defaultweapons.lua:1190]: Same here

File: lua/sim/defaultweapons.lua [hunk 3]
```diff
@@ -493,6 +522,8 @@ DefaultProjectileWeapon = Class(Weapon) {
     end,
 
     -- Create the visual side of rack recoil
+    ---@param self DefaultProjectileWeapon
+    ---@param rackList string
```
Comment (Hdt80bro) [inline lua/sim/defaultweapons.lua:526]: I *think* `rackList` is `RackBoneBlueprint[]`?

**Review by Hdt80bro — COMMENTED** [review]
Looking good!

File: lua/sim/EngineerManager.lua [hunk 4]
```diff
@@ -49,6 +56,11 @@ EngineerManager = Class(BuilderManager) {
     end,
 
     -- Check to see if the unit is buildings something in the category given
+    ---@param unit Unit
+    ---@param econ any
```
Comment (Garanas) [inline lua/sim/EngineerManager.lua:60]: value does not appear to be used

File: lua/sim/EngineerManager.lua [hunk 5]
```diff
@@ -71,6 +83,11 @@ EngineerManager = Class(BuilderManager) {
     end,
 
     -- only pause the assisters of experimentals
+    ---@param unit Unit
+    ---@param econ any
```
Comment (Garanas) [inline lua/sim/EngineerManager.lua:87]: same here

File: lua/sim/FactoryBuilderManager.lua [hunk 6]
```diff
@@ -10,10 +10,18 @@
...
-local AIBuildUnits = import('/lua/ai/aibuildunits.lua')
+
+local TableGetn = table.getn
 
 ---@class FactoryBuilderManager : BuilderManager
 FactoryBuilderManager = Class(BuilderManager) {
+    ---@param self FactoryBuilderManager
+    ---@param brain AIBrain
+    ---@param lType any
+    ---@param location number
```
Comment (Garanas) [inline lua/sim/FactoryBuilderManager.lua:21]: location is usually a `Vector`

File: lua/sim/FactoryBuilderManager.lua [hunk 7]
```diff
@@ -10,10 +10,18 @@
...
+    ---@param radius number
+    ---@param useCenterPoint any
```
Comment (Garanas) [inline lua/sim/FactoryBuilderManager.lua:23]: likely a boolean

File: lua/sim/FactoryBuilderManager.lua [hunk 8]
```diff
@@ -190,11 +230,17 @@ FactoryBuilderManager = Class(BuilderManager) {
...
+    ---@param self FactoryBuilderManager
+    ---@param unit Unit
+    ---@param bType string
     SetupNewFactory = function(self,unit,bType)
         self:SetupFactoryCallbacks({unit}, bType)
         self:ForkThread(self.DelayRallyPoint, unit)
     end,
 
+    ---@param self FactoryBuilderManager
+    ---@param factories any
```
Comment (Garanas) [inline lua/sim/FactoryBuilderManager.lua:242]: list of units

File: lua/sim/FactoryBuilderManager.lua [hunk 9]
```diff
@@ -141,6 +169,10 @@ FactoryBuilderManager = Class(BuilderManager) {
         return units
     end,
 
+    ---@param self FactoryBuilderManager
+    ---@param category string
```
Comment (Garanas) [inline lua/sim/FactoryBuilderManager.lua:173]: aren't these categories, instead of strings?

File: lua/sim/FactoryBuilderManager.lua [hunk 10]
```diff
@@ -252,6 +300,10 @@ FactoryBuilderManager = Class(BuilderManager) {
         self.LocationActive = false
     end,
 
+    ---@param self FactoryBuilderManager
+    ---@param factory any
```
Comment (Garanas) [inline lua/sim/FactoryBuilderManager.lua:304]: a unit

File: lua/sim/FactoryBuilderManager.lua [hunk 11]
```diff
@@ -272,6 +324,9 @@ FactoryBuilderManager = Class(BuilderManager) {
         self:AssignBuildOrder(factory,bType)
     end,
 
+    ---@param self FactoryBuilderManager
+    ---@param factory any
```
Comment (Garanas) [inline lua/sim/FactoryBuilderManager.lua:328]: a unit

Comment (Garanas) [reply lua/sim/FactoryBuilderManager.lua:328]: same for the other references

File: lua/sim/FactoryBuilderManager.lua [hunk 12]
```diff
@@ -393,6 +463,9 @@ FactoryBuilderManager = Class(BuilderManager) {
         end
     end,
 
+    ---@param self FactoryBuilderManager
+    ---@param factory any
+    ---@param finishedUnit any
```
Comment (Garanas) [inline lua/sim/FactoryBuilderManager.lua:468]: a unit

File: lua/sim/MapUtilities.lua [hunk 13]
```diff
@@ -1,17 +1,17 @@
 
 local MarkerUtilities = import("/lua/sim/MarkerUtilities.lua")
-
 local MapResourceCheckApplied = false
 
 --- Attempts to spawn in extractors and hydrocarbons on each marker that is enabled in the game. Attempts 
 -- to ring each extractor with storages and fabricators, similar to how the ringing feature works.
```
Comment (Garanas) [inline lua/sim/MapUtilities.lua:6]: needs a third `-`

File: lua/sim/MapUtilities.lua [hunk 14]
```diff
@@ -52,16 +52,20 @@ function MapResourceCheck()
     local hydro = MarkerUtilities.GetMarkersByType("Hydrocarbon")
 
     -- helper function that attempts to build a unit there
```
Comment (Garanas) [inline lua/sim/MapUtilities.lua:54]: Needs a third `-`, and a capital H for Helper

File: lua/sim/MapUtilities.lua [hunk 15]
```diff
@@ -243,10 +249,10 @@ function iMapToggleRendering()
 
     -- enable the thread if it should not be suspended
     DebugSuspend[type] = not DebugSuspend[type]
-    if not DebugSuspend[type] then 
+    if not DebugSuspend[type] then
```
Comment (Garanas) [inline lua/sim/MapUtilities.lua:252]: all those poor spaces at the end of a line, what did they ever do to you

File: lua/sim/MarkerUtilities.lua [hunk 16]
```diff
@@ -37,6 +37,8 @@ end
 --- Retrieves a single marker on the map.
 -- @param name The name or key of the marker.
 -- returns A marker of the map or nil.
+---@param name string
+---@return any
```
Comment (Garanas) [inline lua/sim/MarkerUtilities.lua:41]: list of markers, not sure if there's a type for that yet. You could construct one based on a _save file of a map

File: lua/sim/MarkerUtilities.lua [hunk 17]
```diff
@@ -141,8 +148,10 @@ local ChainCache = { }
 -- does not exist. This is a shallow copy, which means the
 -- reference is copied but the values are not. If you need a
 -- copy with unique values use GetMarkerByTypeDeep instead.
--- @param type The type of marker to retrieve.
 -- returns A table with markers and its length.
+---@param name string The type of marker to retrieve.
+---@return any
```
Comment (Garanas) [inline lua/sim/MarkerUtilities.lua:153]: list of markers

File: lua/sim/MarkerUtilities.lua [hunk 18]
```diff
@@ -141,8 +148,10 @@ local ChainCache = { }
...
+---@return any
```
Comment (Garanas) [inline lua/sim/MarkerUtilities.lua:154]: number (length of list)

File: lua/sim/MarkerUtilities.lua [hunk 19]
```diff
@@ -181,18 +190,20 @@ end
 -- chain does not exist. This is a deep copy and involves
 -- a lot of additional allocations. Do not use this unless
 -- you strictly need to.
--- @param type The type of marker to retrieve.
 -- returns A table with markers and its length.
+---@param type string The type of marker to retrieve.
+---@return table
+---@return any
```
Comment (Garanas) [inline lua/sim/MarkerUtilities.lua:196]: number (length of list)

File: lua/sim/NukeDamage.lua [hunk 20]
```diff
@@ -20,6 +30,9 @@ NukeAOE = Class() {
         end
     end,
 
+    ---@param self any
+    ---@param instigator any
```
Comment (Garanas) [inline lua/sim/NukeDamage.lua:34]: Unit

File: lua/sim/EngineerManager.lua (hunk 4 above)
Comment (MrRowey) [reply lua/sim/EngineerManager.lua:60]: what should i do in this instant then ?

File: lua/sim/PlatoonFormManager.lua [hunk 21]
```diff
@@ -10,10 +10,15 @@
...
 local AIUtils = import('/lua/ai/aiutilities.lua')
 local Builder = import('/lua/sim/Builder.lua')
-local AIBuildUnits = import('/lua/ai/aibuildunits.lua')
 
 ---@class PlatoonFormManager : BuilderManager
 PlatoonFormManager = Class(BuilderManager) {
+    ---@param self PlatoonFormManager
+    ---@param brain AIBrain
+    ---@param lType any
+    ---@param location number
```
Comment (Garanas) [inline lua/sim/PlatoonFormManager.lua:19]: Vector

File: lua/sim/PlatoonFormManager.lua [hunk 22]
```diff
@@ -173,8 +193,16 @@ PlatoonFormManager = Class(BuilderManager) {
     end,
 }
 
+---@param brain AIBrain
+---@param lType any
+---@param location number
```
Comment (Garanas) [inline lua/sim/PlatoonFormManager.lua:198]: Vector 😄

File: lua/sim/FactoryBuilderManager.lua (hunk 6 above)
Comment (MrRowey) [reply lua/sim/FactoryBuilderManager.lua:21]: Got its same as position then

File: lua/sim/defaultweapons.lua [hunk 23]
```diff
@@ -129,6 +130,9 @@ DefaultProjectileWeapon = Class(Weapon) {
 
     -- This function creates the projectile, and happens when the unit is trying to fire
     -- Called from inside RackSalvoFiringState
+    ---@param self DefaultProjectileWeapon
+    ---@param muzzle string
+    ---@return any
```
Comment (Garanas) [inline lua/sim/defaultweapons.lua:135]: `Projectile`

File: lua/sim/defaultweapons.lua [hunk 24]
```diff
@@ -492,6 +521,8 @@ DefaultProjectileWeapon = Class(Weapon) {
     end,
 
     -- Create the visual side of rack recoil
+    ---@param self DefaultProjectileWeapon
+    ---@param rackList RackBoneBlueprint[]
```
Comment (Garanas) [inline lua/sim/defaultweapons.lua:525]: That type exists?

File: lua/sim/defaultweapons.lua [hunk 25]
```diff
@@ -575,20 +611,25 @@ DefaultProjectileWeapon = Class(Weapon) {
...
 
     -- Sends the weapon to DeadState, probably called by the Owner
+    ---@param self DefaultProjectileWeapon
     OnDestroy = function(self)
         ChangeState(self, self.DeadState)
     end,
 
     -- Checks to see if the weapon is allowed to fire
+    ---@param self DefaultProjectileWeapon
+    ---@return any
```
Comment (Garanas) [inline lua/sim/defaultweapons.lua:621]: boolean

File: lua/sim/defaultweapons.lua [hunk 26]
```diff
@@ -1403,6 +1469,8 @@ DefaultBeamWeapon = Class(DefaultProjectileWeapon) {
     end,
 
     -- Kill the beam if hold fire is requested
+    ---@param self DefaultBeamWeapon
+    ---@param beam any
```
Comment (Garanas) [inline lua/sim/defaultweapons.lua:1473]: CollisionBeam, not sure if that is annotated yet

Comment (Garanas) [reply lua/sim/defaultweapons.lua:1473]: Same for all the other instances

File: lua/sim/defaultweapons.lua [hunk 27]
```diff
@@ -1578,12 +1659,17 @@ DeathNukeWeapon = Class(BareBonesWeapon) {
...
 
+    ---@param self SCUDeathWeapon
     Fire = function(self)
         local bp = self.Blueprint
         local proj = self.unit:CreateProjectile(bp.ProjectileId, 0, 0, 0, nil, nil, nil):SetCollision(false)
         proj:PassDamageData(self:GetDamageTable())
     end,
 }
+
+-- Moved Unsused Imports To bttome for mod support
```
Comment (Garanas) [inline lua/sim/defaultweapons.lua:1674]: See the other PR for a better comment

**Review by Garanas — CHANGES_REQUESTED** [review]
@Hdt80bro feel free to merge as you see fit after the release of tonight.

File: lua/sim/defaultweapons.lua (hunk 24 above)
Comment (MrRowey) [reply lua/sim/defaultweapons.lua:525]: image

Shows up in autofill

File: lua/sim/defaultweapons.lua (hunk 26 above)
Comment (MrRowey) [reply lua/sim/defaultweapons.lua:1473]: I have annotaed CollsionBeam.lua in one of the other Prs

File: lua/sim/MarkerUtilities.lua (hunk 16 above)
Comment (Hdt80bro) [reply lua/sim/MarkerUtilities.lua]: We do--`Marker`!
`/lua/AI/OpAI/BaseManager.lua` has tons of useful stuff

**Pull request merged**